File: Frontend/T-Shirt/src/services/returnService.ts
import { api } from "./api";
import type {
  ReturnEligibilityResponse,
  ReturnResponse,
  CreateReturnPayload,
} from "../types/return";
import type { PageResponse } from "../types/product";

export const returnService = {
  checkEligibility: async (orderId: number): Promise<ReturnEligibilityResponse> => {
    const response = await api.get(`/customer/returns/eligibility/${orderId}`);
    return response.data;
  },

  createReturn: async (payload: CreateReturnPayload): Promise<ReturnResponse> => {
    const response = await api.post("/customer/returns", payload);
    return response.data;
  },

  getUserReturns: async (page = 0, size = 20): Promise<PageResponse<ReturnResponse>> => {
    const response = await api.get("/customer/returns", { params: { page, size } });
    return response.data;
  },

  getReturnById: async (id: number): Promise<ReturnResponse> => {
    const response = await api.get(`/customer/returns/${id}`);
    return response.data;
  },

  cancelReturn: async (id: number): Promise<ReturnResponse> => {
    const response = await api.post(`/customer/returns/${id}/cancel`);
    return response.data;
  },
};

File: Backend/tshirt-store-backend/src/main/java/com/tshirtstore/service/ReturnService.java
package com.tshirtstore.service;

import com.tshirtstore.dto.returnreq.CreateReturnRequest;
import com.tshirtstore.dto.returnreq.ReturnEligibilityResponse;
import com.tshirtstore.dto.returnreq.ReturnResponse;
import com.tshirtstore.entity.QualityCheckResult;
import com.tshirtstore.entity.ReturnStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ReturnService {

    ReturnEligibilityResponse checkEligibility(Long orderId);

    ReturnResponse createReturn(CreateReturnRequest request);

    Page<ReturnResponse> getUserReturns(Pageable pageable);

    ReturnResponse getReturnById(Long returnId);

    ReturnResponse cancelReturn(Long returnId);

    Page<ReturnResponse> getAdminReturns(ReturnStatus status, Pageable pageable);

    ReturnResponse approveReturn(Long returnId);

    ReturnResponse rejectReturn(Long returnId);

    ReturnResponse processQualityCheck(Long returnId, QualityCheckResult result);

    ReturnResponse processRefund(Long returnId);
}
